lib/nostrum/api.ex

```diff
@@ -3853,6 +3879,255 @@ defmodule Nostrum.Api do
+  - `rate_limit_per_user`: Rate limit per user in seconds, can be set to any value in `0..21600`.
+  """
+  @spec start_thread(Channel.id(), thread_without_message_params, AuditLogEntry.reason()) ::
+          {:ok, Channel.t()} | error
+  def start_thread(channel_id, options, reason \\ nil) do
+    request(%{
+      method: :post,
+      route: Constants.thread_without_message(channel_id),
+      body: options,
+      params: [],
+      headers: maybe_add_reason(reason)
+    })
+    |> handle_request_with_decode({:struct, Channel})
+  end
+
+  @doc """
+  Returns a thread member object for the specified user if they are a member of the thread
+  """
+  @spec get_thread_member(Channel.id(), User.id()) :: {:ok, ThreadMember.t()} | error
+  def get_thread_member(thread_id, user_id) do
+    request(:get, Constants.thread_member(thread_id, user_id))
+    |> handle_request_with_decode({:struct, ThreadMember})
+  end
+
+  @doc """
+  Returns a list of thread members for the specified thread.
+
+  This endpoint is restricted according to whether the `GUILD_MEMBERS` privileged intent is enabled.
+  """
+  @spec get_thread_members(Channel.id()) :: {:ok, [ThreadMember.t()]} | error
+  def get_thread_members(thread_id) do
+    request(:get, Constants.thread_members(thread_id))
+    |> handle_request_with_decode({:list, {:struct, ThreadMember}})
+  end
+
+  @doc """
+  Return all active threads for the current guild.
+
+  Response body is a map with the following keys:
+  - `threads`: A list of channel objects.
+  - `members`: A list of `ThreadMemer` objects, one for each returned thread the current user has joined.
+  """
+  @spec list_guild_threads(Guild.id()) ::
+          {:ok, %{threads: [Channel.t()], members: [ThreadMember.t()]}} | error
+  def list_guild_threads(guild_id) do
+    res =
+      request(:get, Constants.guild_active_threads(guild_id))
+      |> handle_request_with_decode
+
+    case res do
+      {:ok, %{threads: channels, members: thread_members}} ->
+        map = %{
+          threads: Util.cast({:list, {:struct, Channel}}, channels),
+          members: Util.cast({:list, {:struct, ThreadMember}}, thread_members)
+        }
+
+        {:ok, map}
+
+      {:error, e} ->
+        {:error, e}
+    end
+  end
+
+  @doc """
+  Returns a list of archived threads for a given channel.
+
+  Threads are sorted by the `archive_timestamp` field, in descending order.
+
+  ## Response body
+  Response body is a map with the following keys:
+  - `threads`: A list of channel objects.
+  - `members`: A list of `ThreadMemer` objects, one for each returned thread the current user has joined.
+  - `has_more`: A boolean indicating whether there are more archived threads that can be fetched.
+
+  ## Options
+  - `before`: Returns threads before this timestamp, can be either a `DateTime` or (ISO8601 timestamp)[`DateTime.to_iso8601/3`].
```

Comment:
I think this link is the wrong way around, isn't it?

```suggestion
  - `before`: Returns threads before this timestamp, can be either a `DateTime` or [ISO8601 timestamp](`DateTime.to_iso8601/3`).
```